lang/java/reef-bridge-java/src/main/java/org/apache/reef/javabridge/generic/JobDriver.java

```diff
@@ -230,41 +234,40 @@ private CLRBufferedLogHandler getCLRBufferedLogHandler() {
   }
 
   private void submitEvaluator(final AllocatedEvaluator eval, final EvaluatorProcess process) {
-    synchronized (JobDriver.this) {
-      eval.setProcess(process);
-      LOG.log(Level.INFO, "Allocated Evaluator: {0}, total running running {1}",
-          new Object[]{eval.getId(), JobDriver.this.contexts.size()});
-      if (JobDriver.this.handlerManager.getAllocatedEvaluatorHandler() == 0) {
-        throw new RuntimeException("Allocated Evaluator Handler not initialized by CLR.");
-      }
-      final AllocatedEvaluatorBridge allocatedEvaluatorBridge =
-          this.allocatedEvaluatorBridgeFactory.getAllocatedEvaluatorBridge(eval, this.nameServerInfo);
-      allocatedEvaluatorBridges.put(allocatedEvaluatorBridge.getId(), allocatedEvaluatorBridge);
-      NativeInterop.clrSystemAllocatedEvaluatorHandlerOnNext(
-          JobDriver.this.handlerManager.getAllocatedEvaluatorHandler(), allocatedEvaluatorBridge, this.interopLogger);
+    eval.setProcess(process);
+    LOG.log(Level.FINE, "Allocated Evaluator: {0}, total running running {1}, time: {2}",
+        new Object[]{eval.getId(), JobDriver.this.contexts.size(), dateFormat.format(new Date())});
+    final long handler = JobDriver.this.handlerManager.getAllocatedEvaluatorHandler();
```

Comment:
How is this guaranteed to be thread safe? There might be multiple threads in this very line at the same time.

Reply:
It is the same handler for multiple callers, but as long as the data inside the handler is thread safe, we should be fine, right? In bridge code, we should ensure all shared data are thread safe. In user's handler, it is user's responsibility to make the handler thread safe.

Reply:
That is my question. Are we sure the bridge code is thread safe?

Reply:
I looked the bridge code all the way to reach to the user's handler, so far I haven't found anywhere that is not safe.

Reply:
Sounds good then :)